### ISSUE TITLE
why are people on 1.0.17 and 1.0.20 in equal measure

### ISSUE THREAD

**Issue by johngruen** [bwinton/SnoozeTabs#382]
![image](https://user-images.githubusercontent.com/3323249/34790941-b4b0a0b8-f643-11e7-8a72-14fa7be13190.png)

**Comment by lmorchard** [comment]
In August 2017, [we switched over to Mozilla signing infrastructure for Test Pilot experiments](https://github.com/bwinton/SnoozeTabs/commit/5751004035d7a006fa56a0f0d5e7d711528bb5fe). I think some URLs changed along the way - i.e. where the .xpi and update.json lived.

I'm fuzzy on the details - but I wonder if we broke updates for folks back on 1.0.17 by changing URLs but not changing the update URL at the old location, pre-signing infrastructure changes? I thought @clouserw fixed that stuff manually, but maybe we missed something?

**Comment by lmorchard** [comment]
Trying to do some more sleuthing on this, but I don't have access to the S3 buckets where Test Pilot keeps add-on builds. I *suspect* the update URLs changed between 1.0.17 and 1.0.18, and we might have left some folks behind. Do we have an old copy of 1.0.17 around on S3? /cc @clouserw

**Comment by wresuolc** [comment]
I found a 1.0.17 copy and it's pointing to the URL I set up to migrate people over to the new stuff, but it's pointing to the new version of 1.0.17.  I assume that people already on 1.0.17 won't get upgraded even if the file is different.  I'll update the redirect to the latest version and I think it will take care of this.

**Comment by wresuolc** [comment]
Actually, it's a little more confusing than that.  https://github.com/bwinton/SnoozeTabs/commit/2cb5408c5a34f163f21f51da98d51277216be6a6 added the original update URL (at the old URL) which worked fine.  A few days later the ID of the add-on was changed https://github.com/bwinton/SnoozeTabs/commit/bbd3fafcedcbb73b09d5fb5b9c3a24676cc9818e .  At some point (maybe the same time) we changed the way test pilot deployed files to use the ID of the add-on in the file path instead of trying to hardcode or parse anything.

So, the automatic deployments have been going to the URL with the new ID but the manifest never got updated.  So, anyone who installed after February 2017 has been on the new versions, anyone before has been stuck on version 1.0.17.  I'll fix the old URL to point to the latest version, but if we push another version we'll have the same problem.  To fix this for real, we actually need to update the old update URL in the manifest.
